### ISSUE TITLE
Implement Basic Reduction Operations

### ISSUE THREAD

**Issue by Mathieu-Boyer** [Mathieu-Boyer/myML#31]
- Sum and mean operations for vectors and matrices
- Foundation for loss functions and neural networks
